src/sourcetransform.rs

```diff
@@ -87,109 +70,242 @@ pub struct Message {
+            watermark: shared::utc_from_timestamp(value.watermark),
+            eventtime: shared::utc_from_timestamp(value.event_time),
         }
     }
 }
+
 #[async_trait]
-impl<T> source_transform_server::SourceTransform for SourceTransformerService<T>
+impl<T> proto::source_transform_server::SourceTransform for SourceTransformerService<T>
 where
     T: SourceTransformer + Send + Sync + 'static,
 {
     async fn source_transform_fn(
         &self,
-        request: Request<SourceTransformRequest>,
-    ) -> Result<Response<SourceTransformResponse>, Status> {
+        request: Request<proto::SourceTransformRequest>,
+    ) -> Result<Response<proto::SourceTransformResponse>, Status> {
         let request = request.into_inner();
 
-        let messages = self
-            .handler
-            .transform(OwnedSourceTransformRequest::new(request))
-            .await;
-
-        let results = messages
-            .into_iter()
-            .map(move |msg| source_transform_response::Result {
-                keys: msg.keys,
-                event_time: prost_timestamp_from_utc(msg.event_time),
-                value: msg.value,
-                tags: msg.tags,
-            })
-            .collect::<Vec<source_transform_response::Result>>();
-
-        Ok(Response::new(SourceTransformResponse { results }))
+        let messages = self.handler.transform(request.into()).await;
+
+        Ok(Response::new(proto::SourceTransformResponse {
+            results: messages
+                .into_iter()
+                .map(move |msg| msg.into())
+                .collect::<Vec<_>>(),
+        }))
     }
 
-    async fn is_ready(&self, _: Request<()>) -> Result<Response<ReadyResponse>, Status> {
-        Ok(Response::new(ReadyResponse { ready: true }))
+    async fn is_ready(&self, _: Request<()>) -> Result<Response<proto::ReadyResponse>, Status> {
+        Ok(Response::new(proto::ReadyResponse { ready: true }))
     }
 }
 
-pub async fn start_uds_server<T>(m: T) -> Result<(), Box<dyn std::error::Error + Send + Sync>>
-where
-    T: SourceTransformer + Send + Sync + 'static,
-{
-    let server_info_file = if std::env::var_os("NUMAFLOW_POD").is_some() {
-        "/var/run/numaflow/server-info"
-    } else {
-        "/tmp/numaflow.server-info"
-    };
-    let socket_file = "/var/run/numaflow/sourcetransform.sock";
-    let listener = shared::create_listener_stream(socket_file, server_info_file)?;
-    let source_transformer_svc = SourceTransformerService { handler: m };
-
-    tonic::transport::Server::builder()
-        .add_service(source_transform_server::SourceTransformServer::new(
-            source_transformer_svc,
-        ))
-        .serve_with_incoming(listener)
-        .await
-        .map_err(Into::into)
+/// gRPC server to start a sourcetransform service
+#[derive(Debug)]
+pub struct Server<T> {
+    sock_addr: PathBuf,
+    max_message_size: usize,
+    server_info_file: PathBuf,
+    sourcetrf_svc: Option<T>,
```

Comment:
I think field name can be generic like `svc` since we don't specify trait bounds for `T` on the struct.